src/components/KSelect/KSelect.vue

```diff
@@ -741,385 +586,162 @@ watch(() => props.items, (newValue, oldValue) => {
   // This prevents the popover from displaying "detached" from the KSelect
   // eslint-disable-next-line @typescript-eslint/ban-ts-comment
   // @ts-ignore
-  if (popper.value && typeof popper.value.updatePopper === 'function') {
+  if (popperElement.value && typeof popperElement.value.updatePopper === 'function') {
     nextTick(() => {
       // eslint-disable-next-line @typescript-eslint/ban-ts-comment
       // @ts-ignore
-      popper.value.updatePopper()
+      popperElement.value.updatePopper()
     })
   }
 }, { deep: true, immediate: true })
 
-const inputWidth = ref(0)
-
-const onPopoverOpen = () => {
-  const inputElem = document.getElementById(selectInputId.value)
-
-  if (inputElem) {
-    inputWidth.value = inputElem.offsetWidth
-  }
-}
-</script>
-
-<style lang="scss" scoped>
-@import '@/styles/tmp-variables';
-@import '@/styles/mixins';
-
-.k-select {
-  width: fit-content; // necessary for correct placement of popup
+watch(filterQuery, () => {
+  emit('query-change', filterQuery.value)
+})
```

Comment:
nit: use the watch value instead of the ref
```suggestion
watch(filterQuery, (q: string) => {
  emit('query-change', q)
})
```